### PR TITLE
add WebSite type to enum

### LINKED ISSUE #2381: Can't add 'WebSite' Monitor type
**Describe Your Environment**
ZoneMinder 1.32.3 in the development dockerfile
Host os is Debian Stretch docker environment is ubuntu xenial

**Describe the bug**
Can't add a 'WebSite' monitor.
I did some investigation by dumping my schema with mysqldump -d Looking at the `Monitors` table it appears the enum for the `type` column is missing the 'WebSite'  type:

```
  `Type` enum('Local','Remote','File','Ffmpeg','Libvlc','cURL') NOT NULL DEFAULT 'Local',
```

This was a recent install so it appears the new install definition script is missing the full list of types for `Monitors`, and `MonitorPresets` tables as the `Controls` and `ControlPresets` contain the full list including 'WebSite'

**To Reproduce**
1. Click add monitor
2. Change source type to 'Web Site'
3. Change function to 'Monitor'
4. Enter web site url and width/height in source.
5. Click save

**Expected behavior**
I expect it to save the monitor, instead the dialog stays up and website does not get saved

**Debug Logs**
```

01/02/19 19:56:42.201600 web_php[20922].ERR [SQL-ERR 'SQLSTATE[01000]: Warning: 1265 Data truncated for column 'Type' at row 1', statement was 'INSERT INTO Monitors SET
 `LinkedMonitors` = NULL, `Name` = 'Monitor-5', `ServerId` = NULL, `StorageId` = '1', `Type` = 'WebSite', `Function` = 'Monitor', Enabled = 1, `RefBlendPerc` = '6', `Al
armRefBlendPerc` = '6', `AnalysisFPSLimit` = NULL, `MaxFPS` = '30', `AlarmMaxFPS` = '30', `Triggers` = '', `Device` = '/dev/video0', `Channel` = '0', `Format` = '255',
`Palette` = '0', `V4LMultiBuffer` = NULL, `V4LCapturesPerFrame` = '1', `Protocol` = NULL, `Host` = NULL, `Port` = '80', Method = '', `Options` = NULL, `SaveJPEGs` = '0'
, `VideoWriter` = '1', `EncoderParameters` = '# Lines beginning with # are a comment \r\n# For changing quality, use the crf option\r\n# 1 is best, 51 is worst quality\
r\n#crf=23', RecordAudio = 0, RTSPDescribe = 0, `LabelFormat` = '%N - %d/%m/%y %H:%M:%S', `LabelX` = '0', `LabelY` = '0', `LabelSize` = '1', `ImageBufferCount` = '20',
`WarmupCount` = '0', `PreEventCount` = '0', `PostEventCount` = '5', `StreamReplayBuffer` = '0', `AlarmFrameCount` = '1', Controllable = 0, `ControlId` = NULL, `ControlD
evice` = NULL, `ControlAddress` = NULL, `AutoStopTimeout` = NULL, TrackMotion = 0, `TrackDelay` = NULL, `ReturnLocation` = '-1', `ReturnDelay` = NULL, `EventPrefix` = '
Event-', `SectionLength` = '600', `FrameSkip` = '0', `MotionFrameSkip` = '0', `AnalysisUpdateDelay` = '0', `FPSReportInterval` = '100', `DefaultView` = 'Events', `Defau
ltRate` = '100', `DefaultScale` = '100', `WebColour` = 'red', Exif = 0, `SignalCheckPoints` = '10', `SignalCheckColour` = '#0000c0', `Path` = '<website address>', `Width` = '640', `Height` = '480', `Refresh` = NULL, Sequence = 4' params:] at includes/database.php line 158
01/02/19 19:56:42.283770 web_php[20922].ERR [Error saving new Monitor.] at includes/actions.php line 568

```

### PR DESCRIPTION
**Pull request by knight-of-ni**
fixes #2381

### CODE REVIEW

**Comment by jacobalberty** [comment]
The zm_update-1.33.1.sql script updates  MonitorPresets and Controls, if I understand the issue correctly its MonitorPresets and Monitors that need updating, Controls looks like it already has the correct enum.

File: db/zm_update-1.33.1.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+-- This updates a 1.33.0 database to 1.33.1
+--
+-- Add WebSite enum to Monitor.Type
+-- Add Refresh column to Monitors table
+--
+
+ALTER TABLE `MonitorPresets` 
+CHANGE COLUMN `Type` `Type` ENUM('Local','Remote','File','Ffmpeg','Libvlc','cURL','WebSite','NVSocket') NOT NULL DEFAULT 'Local' ;
+
+ALTER TABLE `Controls`
```
Comment (connortechnology) [inline db/zm_update-1.33.1.sql:14]: Shouldn't this be Monitors?

Comment (knight-of-ni) [reply db/zm_update-1.33.1.sql:14]: No. The Monitors table was already updated here: https://github.com/ZoneMinder/zoneminder/blob/master/db/zm_update-1.31.43.sql

**Comment by knight-of-ni** [comment]
> The zm_update-1.33.1.sql script updates MonitorPresets and Controls, if I understand the issue correctly its MonitorPresets and Monitors that need updating, Controls looks like it already has the correct enum.

commit fb04ee2 is the one that fixes the issue you described, for new databases. However, commit 53833ad fixes a slightly different problem for dB upgrades (only).

**Comment by knight-of-ni** [comment]
After I made the first commit fixing @2382, I noticed that dB upgrades were not quite right either, so I fixed that as well.

The end result for this PR is to Enum the Website type for the Monitors, MonitorPresets, and Controls dB tables for *both* new databases and upgraded datbases.

**Comment by jacobalberty** [comment]
@knight-of-ni  if I'm upgrading from 1.32.3 is zm_update-1.31.43.sql going to be used?

File: db/zm_update-1.33.1.sql (hunk 1 above)
Comment (jacobalberty) [reply db/zm_update-1.33.1.sql:14]: I just checked doing an zmupdate.pl  going from 1.32.3 with this branch and zm_update-1.31.43.sql doesn't seem to be relevant here as this is the output of `mysqldump -d zm` for the `Monitors` table (up to the `Type` enum)

```
CREATE TABLE `Monitors` (
  `Id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `Name` varchar(64) NOT NULL DEFAULT '',
  `ServerId` int(10) unsigned DEFAULT NULL,
  `StorageId` smallint(5) unsigned DEFAULT '0',
  `Type` enum('Local','Remote','File','Ffmpeg','Libvlc','cURL') NOT NULL DEFAULT 'Local',
```

Comment (connortechnology) [reply db/zm_update-1.33.1.sql:14]: Yeah the current zm_create.sql doesn't have WebSite in the enum.  So we do need it in this update.
While you are at it, can you add the NVSocket type as well?

**Comment by knight-of-ni** [comment]
> if I'm upgrading from 1.32.3 is zm_update-1.31.43.sql going to be used?
You don't need to upgrade if you are simply trying to fix the issue. Just run the script from the command line:
`mysql -uroot -p zm < /path/to/zm_update-1.31.43.sql`

File: db/zm_update-1.33.1.sql (hunk 1 above)
Comment (knight-of-ni) [reply db/zm_update-1.33.1.sql:14]: I set all three Enum's in both the create and update scripts to be: `'Local','Remote','File','Ffmpeg','Libvlc','cURL','WebSite','NVSocket'`

**Pull request merged**